### PR TITLE
allow for instantiation of recording in metric calculator init

### PR DESCRIPTION
**Pull request by alejoe91**
@colehurwitz 

this allows to register the recording_extractor at the __init__ of MetricCalculator

### CODE REVIEW

File: spiketoolkit/validation/metric_calculator.py [hunk 1]
```diff
@@ -881,9 +887,9 @@ def get_metrics_df(self):
                 unit_ids = self.get_unit_ids()
                 for unit_id in all_unit_ids:
                     if unit_id in unit_ids:
-                        all_metrics[unit_id + len(all_unit_ids) * i] = metric[unit_ids.index(unit_id)]
+                        all_metrics[unit_ids.index(unit_id) + len(all_unit_ids) * i] = metric[unit_ids.index(unit_id)]
                     else:
-                        all_metrics[unit_id + len(all_unit_ids) * i] = ''
+                        all_metrics[unit_ids.index(unit_id) + len(all_unit_ids) * i] = ''
```
Comment (alejoe91) [inline spiketoolkit/validation/metric_calculator.py]: this is a fix when unit_ids don't start fom 0

**Comment by colehurwitz** [comment]
I’m not sure this is a good change or not. The recording extractor is already easy to set using the set_recording function and, in isolation, the recording extractor only allows for the user to run SNR

**Comment by colehurwitz** [comment]
You still need to compute quality metric data to do anything other than SNR. I separated relevant data preprocessing for that reason.

**Comment by colehurwitz** [comment]
What convenience does this add? Do you want to compute SNR immediately?

**Comment by alejoe91** [comment]
I think it does no harm. It removes the need of calling the set_recording.

**Comment by colehurwitz** [comment]
Sounds good.

**Pull request merged**